modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -148,13 +148,16 @@ public Object doPost(ExecutionRequest xreq) {
 
         opCtx.setCallback(new OperationCallback() {
 
+            private Object output;
+
             @Override
             public void onChainEnter(OperationType chain) {
                 //
             }
 
             @Override
             public void onChainExit() {
+                setOutput(executionId, (Serializable) output);
```

Comment:
Is it worth doing `output = null` after this to avoid keeping a reference to the output in a field?

Reply:
The operation callback is referenced by the context which is referenced by the thread that handles the chain execution so I'd expect it to be ready for GC once execution is done, no?

Reply:
Yes if it doesn't last longer than that then no need for manual "cleanup".